modules/core_api/fsw/inc/cfe_es_extern_typedefs.h

```diff
@@ -0,0 +1,573 @@
+ * truncated to 32 bits and therefore will not be valid.
+ *
+ * On 64-bit platforms this can be a 64-bit address which will allow the
+ * full memory address in commands and telemetry, but this will break
+ * compatibility with existing control systems, and may also change
+ * the alignment/padding of messages.
+ *
+ * In either case this must be an unsigned type.
+ *
+ * FSW code should access this value via the macros provided, which
+ * converts to the native "cpuaddr" type provided by OSAL.  This macro
+ * provides independence between the message representation and local
+ * representation of a memory address.
+ */
+typedef uint32 CFE_ES_MemAddress_t;
+
+/*
+ * A converter macro to use when initializing an CFE_ES_MemAddress_t
+ * from a pointer value of a different type.
+ *
+ * @note on a 64 bit platform, this macro will truncate the address such
+ * that it will fit into a 32-bit telemetry field.  Obviously, the resulting
+ * value is no longer usable as a memory address after this.
+ */
+#define CFE_ES_MEMADDRESS_C(x) ((CFE_ES_MemAddress_t)((cpuaddr)(x)&0xFFFFFFFF))
+
+/*
+ * Data Sructures shared between API and Message (CMD/TLM) interfaces
+ */
+
+/**
+ * \brief Application Information
+ *
+ * Structure that is used to provide information about an app.
+ * It is primarily used for the QueryOne and QueryAll Commands.
+ *
+ * While this structure is primarily intended for Application info,
+ * it can also represent Library information where only a subset of
+ * the information applies.
+ */
+typedef struct CFE_ES_AppInfo
+{
+    CFE_ResourceId_t ResourceId; /**< \cfetlmmnemonic \ES_APP_ID
+                                      \brief Application or Library ID for this resource */
+    uint32 Type;                 /**< \cfetlmmnemonic \ES_APPTYPE
+                                      \brief The type of App: CORE or EXTERNAL */
+
+    char Name[CFE_MISSION_MAX_API_LEN];       /**< \cfetlmmnemonic \ES_APPNAME
+                                                   \brief The Registered Name of the Application */
+    char EntryPoint[CFE_MISSION_MAX_API_LEN]; /**< \cfetlmmnemonic \ES_APPENTRYPT
+                                                   \brief The Entry Point label for the Application */
+    char FileName[CFE_MISSION_MAX_PATH_LEN];  /**< \cfetlmmnemonic \ES_APPFILENAME
+                                                   \brief The Filename of the file containing the Application */
+
+    CFE_ES_MemOffset_t StackSize;                  /**< \cfetlmmnemonic \ES_STACKSIZE
+                                                        \brief The Stack Size of the Application */
+    uint32 AddressesAreValid;                      /**< \cfetlmmnemonic \ES_ADDRVALID
+                                                        \brief Indicates that the Code, Data, and BSS addresses/sizes are valid */
+    CFE_ES_MemAddress_t CodeAddress;               /**< \cfetlmmnemonic \ES_CODEADDR
```

Comment:
Superficial comment, but there is a repetition of CFE_ES_MemAddress_t with CFE_ES_MemOffset_t, could make this into a new structure to associate these... So instead of code reading 

CFE_ES_MemAddress_t CodeAddress;
CFE_ES_MemOffset_t CodeSize;

could be:
CFE_ES_MemAddOff_t Code;

referenced by Code.Address and Code.Size
THus:

CFE_ES_MemAddOff_t Data;

CFE_ES_MemAddOff_t BSS;

possibly others

Reply:
See https://github.com/nasa/cFE/issues/1300